src/main/kotlin/io/kweb/plugins/staticFiles/StaticFilesPlugin.kt

```diff
@@ -1,29 +1,53 @@
 package io.kweb.plugins.staticFiles
 
 import io.ktor.http.content.files
+import io.ktor.http.content.resources
 import io.ktor.http.content.static
 import io.ktor.http.content.staticRootFolder
+import io.ktor.routing.Route
 import io.ktor.routing.Routing
 import io.kweb.plugins.KwebPlugin
+import io.kweb.plugins.jqueryCore.jqueryCore
 import java.io.File
+import java.net.URL
 
 /**
  * @author rpanic
  *
  * This Plugin serves static files to be used in the frontend
  *
  * @property rootFolder The root folder, where the static assets are saved
+ * @property resourceFolder For serving resources, the path to the folder which will be served
  * @property servedRoute The route where these assets are being served
  */
-class StaticFilesPlugin(private val rootFolder: File, private val servedRoute: String = "assets") : KwebPlugin(){
+class StaticFilesPlugin private constructor(private val servedRoute: String = "assets") : KwebPlugin() {
 
-    override fun decorate(startHead: StringBuilder, endHead: StringBuilder){}
+    private lateinit var datasource: (Route) -> Unit
+
+    constructor(rootFolder: File, servedRoute: String = "assets") : this(servedRoute) {
+        datasource = {
+            it.staticRootFolder = rootFolder
+        }
+    }
+
+    constructor(resourceFolder: ResourceFolder, servedRoute: String = "assets") : this(servedRoute) {
+        datasource = {
+            it.resources(resourceFolder.resourceFolder)
+        }
+    }
+
+    override fun decorate(startHead: StringBuilder, endHead: StringBuilder) {}
 
     override fun appServerConfigurator(routeHandler: Routing) {
-        routeHandler.static(servedRoute){
-            staticRootFolder = rootFolder
+        routeHandler.static(servedRoute) {
+            datasource(this)
```

Comment:
Will this cache the file in memory or will it reload it each time?  The former would probably be preferable for efficiency.

Reply:
Well, I can´t tell for sure. But I have experimented a bit. I enabled the manual CachingHeaders Plugin for Ktor, which does a pretty good job at client-side caching. This works flawelessly and results in access-times of 0ms in most cases. For server-side I can´t really tell. It does get faster from the first to the second access, drops around 50%, but on the other hand, changes to the file in the out folder, result in the changes being reloaded by ktor. If ktor caches the files or not is not clear from my point of view, but i know ktor way too less for that. And after my research I don´t see a way to accomplish this except programming it ourselfes (which could be done but would probably not be the nices solution)